CMakeLists.txt

```
@@ -120,14 +120,19 @@ else ()
 	list(APPEND libobjc_C_SRCS eh_personality.c)
 endif (WIN32)
 
-if (NOT EXISTS "${CMAKE_SOURCE_DIR}/third_party/robin-map/include/tsl/robin_map.h")
-	message(FATAL_ERROR "Git submodules not present, please run:\n\n"
-						" $ git submodule init && git submodule update\n\n"
-						"If you did not checkout via git, you will need to"
-						"fetch the submodule's contents from"
-						"https://github.com/Tessil/robin-map/")
-endif ()
+find_path(ROBIN_MAP_HEADER NAMES "tsl/robin_map.h")
+
+if (NOT ROBIN_MAP_HEADER)
+	if (NOT EXISTS "${CMAKE_SOURCE_DIR}/third_party/robin-map/include/tsl/robin_map.h")
+		message(FATAL_ERROR "Git submodules not present, please run:\n\n"
+							" $ git submodule init && git submodule update\n\n"
+							"If you did not checkout via git, you will need to"
+							"fetch the submodule's contents from"
+							"https://github.com/Tessil/robin-map/")
+	endif ()
 
+	include_directories("${CMAKE_SOURCE_DIR}/third_party/robin-map/include/")
```

```suggestion
	add_library(tsl::robin_map INTERFACE IMPORTED)
	set_target_properties(tsl::robin_map PROPERTIES
	    INTERFACE_INCLUDE_DIRECTORIES "${CMAKE_SOURCE_DIR}/third_party/robin-map/include/")
```

and then it can be linked as above